lib/compress.js

```diff
@@ -1053,7 +1053,7 @@ merge(Compressor.prototype, {
             var val = {};
             for (var i = this.properties.length; --i >= 0;) {
                 var prop = this.properties[i];
-                if (prop instanceof AST_ObjectKeyVal) {
+                if (typeof prop.key === "string") {
```

Comment:
@kzc thanks for bearing with me - I think I get your point now.

Can you check if the tests in the latest commit is good enough to cover this path?